product/cmd/grpc/client/client_test.go

```diff
@@ -0,0 +1,131 @@
+package main
+
+import (
+	"context"
+	"net"
+	"testing"
+
+	"github.com/golang/protobuf/ptypes"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/test/bufconn"
+
+	"product/internal/grpc/domain"
+	"product/internal/grpc/services"
+	repo "product/internal/repository"
+)
+
+type FakeProductRepository struct {
+	store []*domain.Product // temporary storage for products
+}
+
+func (fp *FakeProductRepository) CreateProduct(p *domain.Product) (*domain.Product, error) {
+	// temporary store product
+	fp.store = append(fp.store, p)
+	return p, nil
+}
+
+var fakeRepo = &FakeProductRepository{}
```

Comment:
I've noticed that you're using global variables a lot more recently. Any reason? Best to try avoid them unless absolutely necessary because they add additional places where things can go wrong (i.e. you want to be able to look at code and reason about what it does, but if it can be changed from _anywhere_ you can't be confident in your assessment).

Reply:
Thank you, I was aware of this but I wanted to ensure that the created product was captured in the test